In [16]:
# import pandas and statsmodels
import pandas as pd
import statsmodels.api as sm



# load countries csv data
creditScore = pd.read_csv('Week_4_GP.csv')





In [17]:
# Set X and y data frames 

    # independent variables
X = creditScore[['Credit_Lines','Credit_Length','Number_Missed_Payments','Credit_Utilization','New_Credit','Number_Credit_Types']
]
    # dependent variable
y = creditScore['Credit_Score']

    
    
# Use dtypes to make sure the data is loaded in the right format
creditScore.dtypes



Respondent_ID               int64
Credit_Score                int64
Credit_Lines                int64
Credit_Length               int64
Number_Missed_Payments      int64
Credit_Utilization        float64
New_Credit                  int64
Number_Credit_Types         int64
dtype: object

In [21]:
# Regression with statsmodel

# Use statsmodels to calculate important traits of the regression
# This re-creates the model using the statsmodel package, rather than sklearn
X = sm.add_constant(X)

# Make sure the regression model has an intercept
# the "sm.OLS" command that calculates the regression does not have an intercept by default



# fit the model using OLS
model = sm.OLS(y,X).fit()
 

    
# Summarize the model using .summary() and print the summary 
print_model = model.summary()
print(print_model)


                            OLS Regression Results                            
Dep. Variable:           Credit_Score   R-squared:                       0.485
Model:                            OLS   Adj. R-squared:                  0.472
Method:                 Least Squares   F-statistic:                     36.70
Date:                Thu, 08 Dec 2022   Prob (F-statistic):           3.37e-31
Time:                        04:20:28   Log-Likelihood:                -1364.9
No. Observations:                 241   AIC:                             2744.
Df Residuals:                     234   BIC:                             2768.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                    585

#Questions
1. There are five significant variables
2. Credit_Lines variable is not significant, p=0.462>0.05. Yes, credit lines should not affect credit score.
3. Yes. An increase in the number of missed payments, credit utilization and new credit, the credit score will lower while an increase in the number of credit length and number of credit types would increase the credit score.
4. If an individual misses one additional payment, the credit score will reduce by a coefficient of 31.5457, with all the other factors constant.
5. 48.5% of the variance of the dependent variable being studied is explained by the variance of the independent variables, which is moderate.
6. The adjusted R square can be used to improve the model since it takes care of the additional variables. The adjusted R square can only increase if the new term enhances the model above what would be obtained by probability and decreases when a predictor enhances the model less than what is predicted by chance.

In [19]:
# Import the Time Series data set
creditScorePersonal = pd.read_csv('Week_4_GP_Time_Series.csv')



# Set an Xpersonal using the same variables as before, plus Q2, Q3, and Q4

Xpersonal = creditScorePersonal[['Q2','Q3','Q4','Credit_Lines','Credit_Length','Number_Missed_Payments','Credit_Utilization','New_Credit','Number_Credit_Types']]

# Set ypersonal to your Credit Score over time
ypersonal = creditScorePersonal['Credit_Score']


# use .head(10) to get a feel for your credit score data. Try to see if it makes intuitive sense at a glance
creditScorePersonal.head(10)


Year Quarter  Q2  Q3  Q4  New_Credit  Credit_Lines  Credit_Length  \
0  2014      Q1   0   0   0           1             1              1   
1  2014      Q2   1   0   0           1             1              1   
2  2014      Q3   0   1   0           0             1              1   
3  2014      Q4   0   0   1           1             2              1   
4  2015      Q1   0   0   0           1             2              1   
5  2015      Q2   1   0   0           1             2              2   
6  2015      Q3   0   1   0           1             2              2   
7  2015      Q4   0   0   1           0             2              2   
8  2016      Q1   0   0   0           0             2              2   
9  2016      Q2   1   0   0           0             2              3   

   Number_Missed_Payments  Credit_Utilization  Number_Credit_Types  \
0                       0                0.38                    1   
1                       0                0.34                    1   
2                       0                0.14                    1   
3                       0                0.19                    1   
4                       0                0.20                    1   
5                       0                0.56                    1   
6                       1                0.70                    1   
7                       1                0.62                    1   
8                       1                0.60                    1   
9                       1                0.58                    1   

   Credit_Score  
0           655  
1           653  
2           707  
3           614  
4           664  
5           647  
6           592  
7           603  
8           628  
9           639

In [20]:
# Regression with statsmodel

# Use statsmodels to calculate important traits of the regression
# This re-creates the model using the statsmodel package, rather than sklearn
Xpersonal = sm.add_constant(Xpersonal)

# Make sure the regression model has an intercept
# the "sm.OLS" command that calculates the regression does not have an intercept by default

model = sm.OLS(ypersonal,Xpersonal).fit()

# fit the model using OLS
print_model = model.summary()
print(print_model)
 

    
# Summarize the model using .summary() and print the summary 



                            OLS Regression Results                            
Dep. Variable:           Credit_Score   R-squared:                       0.901
Model:                            OLS   Adj. R-squared:                  0.857
Method:                 Least Squares   F-statistic:                     20.27
Date:                Thu, 08 Dec 2022   Prob (F-statistic):           3.49e-08
Time:                        02:09:45   Log-Likelihood:                -119.41
No. Observations:                  30   AIC:                             258.8
Df Residuals:                      20   BIC:                             272.8
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                    668

#Questions
7. 7 variables are significant.
8. Q2 and Q3.
9. Yes. An increase in the number of missed payments, credit utilization, credit lines, new credit and Q4 variable, the credit score will lower while an increase in the number of credit length and number of credit types would increase the credit score.
10. 90.1% of the variance of the dependent variable being studied is explained by the variance of the independent variables, which means the explained variation is high out of the total variation.
11. With a control of one of the variables, the effect of Credit Lines variable is increased, and thus becomes significant.